### ISSUE TITLE
FIXME: 0.7 does not have menus?

### ISSUE THREAD

**Issue by Dukko** [RimSort/RimSort#377]
**Describe the bug**
I updated to 0.7 and now all my menus are gone lol. I can see the modlist and the sorting buttons, but can't configure anything or run todds. Any idea why?

**To Reproduce**
update on linux Arch from 0.6 to 0.7

**Expected behavior**
Have access to menus 

**Screenshots**
https://i.imgur.com/1yt68UY.png

**Please complete the following information:**
 - OS: Linux Arch
 - RimSort version: 0.7
 - Please provide a `RimSort.log`:
 - Are you running from Python interpreter or from a compiled RimSort release? compiled release

**Comment by Limfjorden** [comment]
It's all loacated in settings now.

**Comment by Dukko** [comment]
Right, so how do I run todds?

**Comment by Chunnyluny** [comment]
> Right, so how do I run todds?

Go to textures and use optimize textures, there are only 2 options, in the settings menu you can choose to only optimize active mods or not, if you have different mod list for different runs, you can just optimize both inactive and active mods.

**Comment by Dukko** [comment]
https://i.imgur.com/YD6yARm.png I don't see a way to run it though.

**Comment by hk21702** [comment]
It does seem like the menu bar is missing. I'm not entirely sure why, seems like an Arch issue since it works fine in Ubuntu and the other strictly supported OSes. cc @twstagg 

For context, this is expected behavior to find todds. (This is Windows 11)
![image](https://github.com/RimSort/RimSort/assets/28567881/e44302e6-43f8-4417-b304-c5738482c432)

**Comment by hk21702** [comment]
If you are using KDE, this thread may be of interest to you. https://bbs.archlinux.org/viewtopic.php?id=286140

I personally don't use Arch and don't use Linux as a daily driver so I'm not entirely sure if it is relevant, but just throwing it out there just in case.

**Comment by Mentalgen** [comment]
The solution was indeed within the above thread, typing in the command "kcmshell6 kcm_kded" and find "Application menus daemon". Disabling this service returns the global menu to the top of RimSort.

I do wonder why this is required, there's other applications that's able to use the global menu on the top bar....

**Comment by hk21702** [comment]
It is unlikely that this issue will really get looked at since I don't think anyone on the dev team uses Arch/KDE and this issue appears to be specific to that. We only explicitly build on Ubuntu runners, but I will leave this issue open in case a better solution comes up.

**Comment by Dukko** [comment]
Yep, the issue was resolved by disabling the global menu applet in KDE. Odd. Thanks!

**Comment by ChaoticTabris** [comment]
Issue is not exclusive to Arch. Observed the same issue on Nobara (which is based on Fedora). It probably relates to KDE + Wayland.

**Comment by hk21702** [comment]
As a workaround, maybe we can detect the DE being used, and if it is KDE, force disable using the native menu bar. The issue doesn't seem to exist with MacOS or gnome where the native menu bar works fine.

**Comment by hk21702** [comment]
After poking around for way too long, I think this is indeed an issue with kde + wayland. Can one of you try falling back to x11? If it is that, then the only solution I can think of is implementing something to disable using the global native menu bar for KDE, or telling people to use x11 and yell at wayland people.

https://bugs.kde.org/show_bug.cgi?id=424485
https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/52
https://bugs.kde.org/show_bug.cgi?id=483075

**Comment by hk21702** [comment]
Can someone with this issue confirm if it still happens when running from the Python interpreter? I.e., not a compiled release. The releases we provide are specifically compiled/built using Ubuntu runners from GitHub. As a result, different distros, or even flavors of Ubuntu may cause issues if using the pre-compiled releases.

**Comment by twstagg** [comment]
> Can someone with this issue confirm if it still happens when running from the Python interpreter? I.e., not a compiled release. The releases we provide are specifically compiled/built using Ubuntu runners from GitHub. As a result, different distros, or even flavors of Ubuntu may cause issues if using the pre-compiled releases.

Hi @Dukko . Thank you for your work on your Stellaris modpack and other things. I've had many hours of enjoyment due to you. I was wondering, if you would be able to test this for us? It would really help us to determine how to best "wrap up" this issue.

Thanks again!

twsta

**Comment by Mentalgen** [comment]
I've tested again and the issue is still present on the latest commit (762c06e).

**Comment by hk21702** [comment]
@Mentalgen this is through the python interpreter, right? Not a compiled release?

**Comment by Mentalgen** [comment]
Correct

**Comment by KillerGods65** [comment]
Hi, i just wan to add a little more... i use x11 on manjaro kde, i have less than a week since i change from win to linux and my knowledge in programing is low, so i dont know what is that python interpreter, i used first a git clone, don't know if is that; and the menus did not appear so i try using the ubuntu release and neither there, although the solution of disabling the "Application menus daemon" works for me too, I'm willing to help test things, but you are going to have to explain to me how to do them haha
![imagen](https://github.com/user-attachments/assets/d370c148-1a47-4eb2-a374-f220b19de9a6)

**Comment by LionelColaso** [comment]
@KillerGods65 you can just follow [build guide](https://rimsort.github.io/RimSort/development-guide/development-setup/#introduction) its easy and see if ur issue gets resolved u can also msg here or on discord if u run into issue

**Comment by KillerGods65** [comment]
@LionelColaso Hi, i followed the build guide you told me, and i still have the problem, so i think is a interaction between rimsort and this KDE "application menu daemon"

this one is just after building
![imagen](https://github.com/user-attachments/assets/316f87aa-2b6b-4d52-8520-82bc3bc5180b)

this one is after disabling the "application menu daemon" 
![imagen](https://github.com/user-attachments/assets/b34c92c3-0d9f-4a72-af7c-89fc41cd9eb1)

as i have said before, im really new to linux so this is on a almost clean install of manjaro kde

**Comment by twstagg** [comment]
> Yep, the issue was resolved by disabling the global menu applet in KDE. Odd. Thanks!

Closing this out due to the resolution being stated in the Issue above by Dukko. Please use this workaround if you encounter this in the future!
